SystemDocumentation/SystemDocs.txt

```diff
@@ -0,0 +1,206 @@
+the database.
+
+We have our Update queries, these queries don't insert any new rows into the database, but they will update existing rows in the
+database.
+
+DatabaseUrlGenerator.py -
+Overview: This file will get the database url for local development and for production development so we don't have that in our codebase
+
+Use Cases:
+getLines(): This will get database info from a local text file, only used for local development
+
+getDatabaseUrl(): This will grab the database url locally text file and is used for local development
+
+DbConnection.py -
+Overview: This is the code that handles the connecting to the database. This is a pretty simple interaction.
+This class is only used in the Database.py class, it allows for that class to connect to the database on Heroku.
+
+Use Cases:
+There are only 2 use cases for this class:
+
+readQuery, this function is used for the get queries in Database.py, it gives us information back from the database in python List form.
+
+writeQuery, this function doesn't return anything because it is used for insert and update queries in Database.py.
+
+These are only supposed to be used in Database.py class.
+
+
+DatabaseUtils.py -
+Overview: This is a file that holds a lot of logic and utilities away from the queries and the main file.
+
+Use Cases:
+
+getTotalPrice, this will get the total price for the items in the users cart it is used in the main.py file.
+
+newCustomer, this will write a name and an email to the database.
+
+newTransaction, this will write a new transaction to the database with a random transaction id associated with it.
+
+listToString, this function will take the list of items from the cart and make them a list for transactions inserting.
+
+createSale, this function will create a sale from the manager page and do the appropriate things in the database.
+
+moneyMadeFromProduct, this function will calculate the money made from each product in your inventory. Available from the manager page.
+
+generateReport, this will create a report of all the items in the database with the moneyMadeFromProduct on them.
+
+purchasedItems, this function will subtract purchased items from the inventory_log when someone buys something from the inventory.
+
+addToSearchQuery, this adds things to the cashiers search query.
+
+addToManagerSearchQuery, this will do the same thing as above but for the manager only.
+
+main.py -
+Overview: This is the brain of our program. It handles all the internet requests, and handles all of the routes for the system.
+All of the program connects to the main.
+
+Use Cases: Its hard to gauge use cases for this function since none of its functions can really be called but we will still go over what they do.
+We will be covering what the routes do.
+
+@app.route("/managerPage")
+This route will take us to the manager page and will render that page and pass all necessary inputs in.
+
+@app.route("/searchInventoryById", methods=["POST"])
```

Comment:
Now that I think about it, I'm just going to leave them in. I don't feel like doing much code cleaning...